Base.rte/Devices/Tools/GrappleGun/Grapple.lua

```diff
@@ -54,64 +54,73 @@ function Create(self)
-			self.parent:GetController():SetState(Controller.AIM_SHARP, false);
-		end
-
-		local negNum = self.parent.HFlipped and -1 or 1;
-
+		
 		self.ToDelete = false;
 		self.ToSettle = false;
 
-		self.pieSelection = 0;
-		if self.parentGun ~= nil and self.parentGun.ID ~= 255 then
+		self.lineVec = SceneMan:ShortestDistance(self.parent.Pos, self.Pos, self.mapWrapsX);
+		self.lineLength = self.lineVec.Magnitude;
+
+		if self.parentGun and self.parentGun.ID ~= 255 then
+			self.parent = ToMOSRotating(MovableMan:GetMOFromID(self.parentGun.RootID));
+
+			if self.parentGun.Magazine then
+				self.parentGun.Magazine.Scale = 0;
+			end
 			startPos = self.parentGun.Pos;
-			
-			self.pieSelection = self.parentGun.Sharpness;
-			self.parentGun.Vel = Vector(0, -1);
+			-- Rotate the gun towards the hook
+			local flipAng = self.parent.HFlipped and 3.14 or 0;
+			self.parentGun.RotAngle = self.lineVec.AbsRadAngle + flipAng;
+			if self.parentGun.Sharpness ~= 0 then
+				self.pieSelection = self.parentGun.Sharpness;
+				self.parentGun.Sharpness = 0;
+			end
 			if self.parentGun.FiredFrame then
 				if self.actionMode == 1 then
 					self.ToDelete = true;
 				else
 					self.canRelease = true;
 				end
 			end
-			
-			self.parentGun.StanceOffset = Vector(12, 1);
-			self.parentGun.SharpStanceOffset = Vector(12, 1);
-			
-			if self.parentGun.FiredFrame and self.canRelease == true and (Vector(self.parentGun.Vel.X, self.parentGun.Vel.Y) ~= Vector(0, -1) or self.parentGun:IsActivated()) then
+			if self.parentGun.FiredFrame and self.canRelease and (Vector(self.parentGun.Vel.X, self.parentGun.Vel.Y) ~= Vector(0, -1) or self.parentGun:IsActivated()) then
 				self.ToDelete = true;
 			end
 		end
-		
-		self.lineVec = SceneMan:ShortestDistance(self.parent.Pos, self.Pos, self.mapWrapsX);
-		self.lineLength = self.lineVec.Magnitude;
-
+		if IsAHuman(self.parent) then
+			self.parent = ToAHuman(self.parent);
+			-- We now have a user that controls this grapple
+			controller = self.parent:GetController();
+			-- Point the gun towards the hook if our user is holding it
+			if (self.parentGun and self.parentGun.ID ~= 255) and (self.parentGun:GetRootParent().ID == self.parent.ID) then
+				local offset = Vector(ToMOSprite(self.parentGun:GetParent()):GetSpriteWidth(), 0):RadRotate(self.parent.FlipFactor * (self.lineVec.AbsRadAngle - self.parent:GetAimAngle(true)))
+				self.parentGun.StanceOffset = offset;
+				if self.parent.EquippedItem and self.parent.EquippedItem.ID == self.parentGun.ID and (self.parent.Vel.Magnitude < 5 and controller:IsState(Controller.AIM_SHARP)) then
+					self.parentGun.RotAngle = self.parent:GetAimAngle(false) * self.parentGun.FlipFactor;
+					startPos = self.parent.Pos;
+				else
+					self.parentGun.SharpStanceOffset = offset;
+				end
+			end
+			-- Prevent the user from spinning like crazy
+			if self.parent.Status > 0 then
+				self.parent.AngularVel = self.parent.AngularVel / (1 + math.abs(self.parent.AngularVel) * 0.01);
```

Comment:
Well fuck, obviously the unnecessary comments were left there since you skimmed it last time. Will fix later.

The AngularVel nerf thing is more exponential and it does make a difference, trust me.

Reply:
Okay, sounds good.